src/field.cpp

```diff
@@ -1038,6 +1139,25 @@ bool map::process_fields_in_submap( submap *const current_submap,
                         }
 
                         // Consume adjacent fuel / terrain / webs to spread.
+                        // Allow raging fires (and only raging fires) to spread up
+                        // Spreading down is achieved by wrecking the walls/floor and then falling
+                        if( zlevels && cur->getFieldDensity() == 3 && p.z < OVERMAP_HEIGHT ) {
```

Comment:
Aah, one piece missing, which is turning destroyed terrain into open air instead of rubble or similar if there is open space below it.  Not insisting on that for this PR, but it'd be awesome :D